src/js/utils/MesosStateUtil.js

```diff
@@ -2,8 +2,34 @@ import Util from './Util';
 
 const RESOURCE_KEYS = ['cpus', 'disk', 'mem'];
 
+function setSchedulerName(name, tasks) {
+  return tasks.map(function (task) {
+    let newTask = Object.assign({}, task);
+
+    newTask.scheduler = name;
+
+    return newTask;
+  });
+}
+
 const MesosStateUtil = {
 
+  assignSchedulerNameToTasks(state) {
```

Comment:
I would prefer that we introduce the suggested label to tell whether it has been started by the root Marathon.
Let's tackle the other scenario when that PR comes up. It is hard for me to review that need until I understand what that task is about.

The reason I don't like the name, is that it feels like duplicating information from the framework (id and name) instead of having that one identifier + helpful properties that works for the use cases.

I don't see what else you need the name for that you can't use the `framework_id` for.